### DISCUSSION TITLE
Support template language escaping when generating classes

### DISCUSSION THREAD

**Discussion by skttl** [tailwindlabs/tailwindcss#12036] [Ideas]
This is probably mostly for the container queries plugin, but I'm doing frontend in ASP.net Core (razor cshtml), where the @-sign is used to put in variables, like `<h1>@headline</h1>`.

This makes it hard to use the container queries plugin, because it requires classes and variants to be prefixed with @, like `<div class="@container"><div class="bg-red @sm:bg-blue"></div></div>`

To do this in razor cshtml, I need to either put the value in a string, or escape the @-sign, so either
`<div class="@("@container")"><div class="@("bg-red @sm:bg-blue")"></div></div>` or `<div class="@@container"><div class="bg-red @@sm:bg-blue"></div></div>`

I prefer the last one, it's much easier to write. But it makes tailwindcss unable to find the @container and @sm:bg-blue classes, because the @'s are doubled.

Is there any way the class finder can be made aware of this, so in cshtml files a double @@ is counted as a @. I guess in PHP there would be a similar problem with $'s.

I've also suggested making the @ configurable in the plugin, but I think having escape-aware class finding is the better option. https://github.com/tailwindlabs/tailwindcss-container-queries/issues/22

**Comment by wongjn** [comment]
You could consider configuring a [content transformer](https://tailwindcss.com/docs/content-configuration#transforming-source-files) that replaces `@@` with `@`.

**Reply by skttl** [reply]
Thanks, just found that possibility - that should do it, and is probably the best option :)

**Reply by sitefinitysteve** [reply]
@skttl I'm hitting this now, is there some setting to get Razor to understand the @@ is escaped?

![image](https://github.com/tailwindlabs/tailwindcss/assets/1542376/3901d07b-7cc2-4116-8bfb-76470e349d88)
